src/parser/mod.rs

```diff
@@ -6793,6 +6977,13 @@ impl<'a> Parser<'a> {
                 Keyword::ARRAY => {
                     if dialect_of!(self is SnowflakeDialect) {
                         Ok(DataType::Array(ArrayElemTypeDef::None))
+                    } else if dialect_of!(self is ClickHouseDialect) {
+                        self.expect_token(&Token::LParen)?;
+                        let (inside_type, _) = self.parse_data_type_helper()?;
```

Comment:
```suggestion
                        let (inside_type, _) = self.parse_data_type()?;
```

Reply:
I think this one might no longer be needed per the `parse_sub_type` comment earlier, if that can be reused here

Reply:
I updated parsing `Array` for ClickHouse with `parse_comma_separated` function.